### PR TITLE
feat(graphql): Add sorting option to GraphQL search

### PR DESCRIPTION
**Pull request by nmbryant**
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

**Comment by nmbryant** [comment]
If there are any tests related to this, please let me know. I took a look and couldn't find tests related to the GraphQL queries.
Also, please let me know if there are any particular docs that need to be updated

**Comment by jjoyce0510** [comment]
Thanks for the PR @nmbryant! Will be taking a look this week.

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/search/SearchResolver.java [hunk 1]
```diff
@@ -45,17 +48,20 @@
 
     return CompletableFuture.supplyAsync(() -> {
       try {
-        log.debug("Executing search. entity type {}, query {}, filters: {}, start: {}, count: {}", input.getType(),
-            input.getQuery(), input.getFilters(), start, count);
+        log.debug("Executing search. entity type {}, query {}, filters: {}, start: {}, count: {}, sort: {}", input.getType(),
```
Comment (jjoyce0510) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/search/SearchResolver.java:51]: Recently, we've migrated to using the SearchAcrossEntitiesResolver from the UI. Are you intending to use the API from the UI or from the API?

**Review by jjoyce0510 — COMMENTED** [review]
Overall this looks great! 

But one question: What are you hoping to achieve with this PR? 

Unfortunately, we've recently moved away from using the "search" API for the UI search functionality. Instead we use a multi-entity search resolver called searchAcrossEntities. 

If you are intending to sort the UI search results, we will need to change that resolver with the help of @dexter-mh-lee. 

If you are intending to access the single-entity search API via a API directly, then this mostly looks great!

**Comment by nmbryant** [comment]
Our use case with DataHub plans on using a couple of single-entity searches so we wanted to add sorting on the GraphQL side rather than do the sorting on the UI side. I do need to take a further look at searchAcrossEntities, but I think the single entity search is what we will want to use.

**Comment by nmbryant** [comment]
@jjoyce0510 Is this something that you want merged upstream? If it is, I can resolve the merge conflicts and make changes based on any feedback you may have. Otherwise, I will just copy the changes into our fork and close this PR.

**Comment by jjoyce0510** [comment]
@nmbryant We'd definitely be interested in taking this upstream, yes! If you're able to clear the conflicts, we can work to quickly get it in

**Comment by nmbryant** [comment]
Awesome, I have the merge conflicts resolved locally just need to do some testing tomorrow before it's ready to go.

**Comment by nmbryant** [comment]
@jjoyce0510 Fixed the merge conflicts and did some testing, it's ready for review

**Comment by jjoyce0510** [comment]
@nmbryant Hi!

I know there's been some time since there was last activity on this PR!

If you're still interested and willing to make these changes, we'll need to update to get the more recent changes. There should be some improvements that will make this PR much easier.

Otherwise, we will look at taking this over. Apologies for the delay on our side. 

John

**Comment by aditya-radhakrishnan** [comment]
Closing this PR due to inactivity. @nmbryant if you'd like to open this up again, please let us know!

**Pull request closed without merging**